proxy/server/server.go

```diff
@@ -312,6 +380,22 @@ func (s *T) newTLSCfg(hostCfgs map[engine.HostKey]engine.Host) (*tls.Config, err
 	}
 
 	config.BuildNameToCertificate()
+
+	//Lock down this specific getCertFuncs map
```

Comment:
Here and in many other places `//` should be separated from the text with a space just to be consistent.